packages/flutter/lib/src/widgets/form.dart

```diff
@@ -15,6 +15,56 @@ import 'will_pop_scope.dart';
 /// descendant of this [Form]. To obtain the [FormState], you may use [Form.of]
 /// with a context whose ancestor is the [Form], or pass a [GlobalKey] to the
 /// [Form] constructor and call [GlobalKey.currentState].
+///
+/// {@tool snippet --template=stateful_widget_material}
+/// This example shows a [Form] with one [TextFormField] and a [RaisedButton]. A
+/// [GlobalKey] is required to identify the [Form], and it is used here to
```

Comment:
this sounds like you always have to provide a global key, but we’re only giving it one her so we can access the forms state for validation purposes, right?

Reply:
Yeah, as I understand it, the key is always required. Using it to validate the form is just an example. Have I misunderstood?